a3p-integration/proposals/z:acceptance/vaults.test.js

```diff
@@ -0,0 +1,250 @@
+import test from 'ava';
+import {
+  agops,
+  agoric,
+  bankSend,
+  getUser,
+  openVault,
+  adjustVault,
+  closeVault,
+  getISTBalance,
+  getContractInfo,
+  ATOM_DENOM,
+  USER1ADDR,
+  waitForBlock,
+} from '@agoric/synthetic-chain';
+import { getBalance } from './test-lib/utils.js';
+
+test.serial('attempt to open vaults under the minimum amount', async t => {
+  const activeVaultsBefore = await agops.vaults('list', '--from', USER1ADDR);
+  await bankSend(USER1ADDR, `20000000${ATOM_DENOM}`);
+  t.log('active vaults before:', activeVaultsBefore);
+
+  const mint = '3.0';
+  const collateral = '5.0';
+  const error = await t.throwsAsync(() =>
+    openVault(USER1ADDR, mint, collateral),
+  );
+
+  t.true(
+    error?.message.includes(
+      'Error: Vault creation requires a minInitialDebt of {"brand":"[Alleged: IST brand]","value":"[5000000n]"}',
+    ),
+    'Error message does not contain the expected text',
+  );
+
+  const activeVaultsAfter = await agops.vaults('list', '--from', USER1ADDR);
+  t.log('active vaults after:', activeVaultsAfter);
+
+  t.true(
+    activeVaultsAfter.length === activeVaultsBefore.length,
+    'The number of active vaults should remain the same.',
+  );
+});
+
+test.serial('open new vault', async t => {
+  const istBalanceBefore = await getISTBalance(USER1ADDR);
+  const activeVaultsBefore = await agops.vaults('list', '--from', USER1ADDR);
+  t.log('uist balance before:', istBalanceBefore);
+  t.log('active vaults before:', activeVaultsBefore);
+
+  await bankSend(USER1ADDR, `20000000${ATOM_DENOM}`);
+
+  const mint = '5.0';
+  const collateral = '10.0';
+  await openVault(USER1ADDR, mint, collateral);
+
+  const istBalanceAfter = await getISTBalance(USER1ADDR);
+  const activeVaultsAfter = await agops.vaults('list', '--from', USER1ADDR);
+  t.log('uist balance after:', istBalanceAfter);
+  t.log('active vaults after:', activeVaultsAfter);
+
+  t.is(
+    istBalanceBefore + 5,
+    istBalanceAfter,
+    'The IST balance should increase by the minted amount',
+  );
+  t.is(
+    activeVaultsAfter.length,
+    activeVaultsBefore.length + 1,
+    `The number of active vaults should increase after opening a new vault.`,
+  );
+});
+
+test.serial('remove collateral', async t => {
+  const activeVaults = await agops.vaults('list', '--from', USER1ADDR);
+  const vaultPath = activeVaults[activeVaults.length - 1];
+  const vaultID = vaultPath.split('.').pop();
+
+  let vaultData = await getContractInfo(vaultPath, { agoric, prefix: '' });
+  const collateralBefore = vaultData.locked.value;
+  t.log('vault collateral before:', collateralBefore);
+
+  await adjustVault(USER1ADDR, vaultID, { wantCollateral: 1.0 });
+  await waitForBlock();
+
+  vaultData = await getContractInfo(vaultPath, { agoric, prefix: '' });
+
+  const collateralAfter = vaultData.locked.value;
+  t.log('vault collateral after:', collateralAfter);
+
+  t.is(
+    Number(collateralBefore),
+    Number(collateralAfter) + 1000000,
```

Comment:
BigInts compare just fine. Please don't convert to `Number` unnecessarily.
```suggestion
    collateralBefore,
    collateralAfter + 1_000_000n,
```

(and similar tests later.)

Reply:
Changes addressed on commit [2afeb25](https://github.com/Agoric/agoric-sdk/pull/10123/commits/2afeb253b4fc99adcd5ca68b378ef31f37f652a0)